src/pkg/zoci/copier.go

```diff
@@ -35,5 +47,27 @@ func CopyPackage(ctx context.Context, src *Remote, dst *Remote, include func(d o
 		return err
 	}
 	progressBar.Successf("Copied %s", src.Repo().Reference)
+
+	srcRoot, err := src.ResolveRoot(ctx)
+	if err != nil {
+		return err
+	}
+
+	b, err := srcManifest.MarshalJSON()
+	if err != nil {
+		return err
+	}
+	expected := content.NewDescriptorFromBytes(ocispec.MediaTypeImageManifest, b)
+
+	if err := dst.Repo().Manifests().PushReference(ctx, expected, bytes.NewReader(b), srcRoot.Digest.String()); err != nil {
+		return err
+	}
+
+	tag := src.Repo().Reference.Reference
+	if err := dst.UpdateIndex(ctx, tag, expected); err != nil {
+		return err
+	}
+
+	message.Infof("Published %s to %s", src.Repo().Reference, dst.Repo().Reference)
```

Comment:
should `*Remote` also have a `r.log`? food for thought

Reply:
I think it's a fair question, zoci will also be used as a library by uds cli. Let's ask the UDS CLI team how they log and if they want their own user experience around logging.

Maybe we could also do some interesting things with unit testing if we passed in our logger, though that's a whole other conversation